### ISSUE TITLE
🐛 Bug Report: CSSOM.parse is not a function

### ISSUE THREAD

**Issue by magnusp** [backstage/backstage#16388]
### 📜 Description

Starting from a fresh Backstage app warnings are reported when running tests

### 👍 Expected behavior

Warning should not be emitted

### 👎 Actual Behavior with Screenshots

```
> yarn test
...
console.warn
      Warning: [JSS] CSSOM.parse is not a function

      at Object.warning (../../../node_modules/tiny-warning/dist/tiny-warning.cjs.js:13:15)
      at insertRule (../../../node_modules/jss/dist/jss.cjs.js:1812:72)
      at DomRenderer._insertRule [as insertRule] (../../../node_modules/jss/dist/jss.cjs.js:1966:22)
      at DomRenderer.insertRule [as insertRules] (../../../node_modules/jss/dist/jss.cjs.js:1926:12)
      at DomRenderer.insertRules [as insertRule] (../../../node_modules/jss/dist/jss.cjs.js:1958:12)
      at DomRenderer.insertRule [as insertRules] (../../../node_modules/jss/dist/jss.cjs.js:1926:12)
      at DomRenderer.insertRules [as deploy] (../../../node_modules/jss/dist/jss.cjs.js:1913:12)
      at StyleSheet.deploy (../../../node_modules/jss/dist/jss.cjs.js:1243:38)
      at StyleSheet.deploy (../../../node_modules/jss/dist/jss.cjs.js:1088:30)
      at attach (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:139:18)
      at ../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:258:7
      at useSynchronousEffect (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:211:14)
      at useStyles (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:250:5)
      at useStyles$l (../../../node_modules/@backstage/core-components/dist/index.esm.js:3770:19)
      at renderWithHooks (../../../node_modules/react-dom/cjs/react-dom.development.js:14985:18)
      at mountIndeterminateComponent (../../../node_modules/react-dom/cjs/react-dom.development.js:17811:13)
      at beginWork (../../../node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at beginWork$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:23940:14)
      at performUnitOfWork (../../../node_modules/react-dom/cjs/react-dom.development.js:22779:12)
      at workLoopSync (../../../node_modules/react-dom/cjs/react-dom.development.js:22707:5)
      at renderRootSync (../../../node_modules/react-dom/cjs/react-dom.development.js:22670:7)
      at performSyncWorkOnRoot (../../../node_modules/react-dom/cjs/react-dom.development.js:22293:18)
      at ../../../node_modules/react-dom/cjs/react-dom.development.js:11327:26
      at unstable_runWithPriority (../../../node_modules/scheduler/cjs/scheduler.development.js:468:12)
      at runWithPriority$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:11276:10)
      at flushSyncCallbackQueueImpl (../../../node_modules/react-dom/cjs/react-dom.development.js:11322:9)
      at flushSyncCallbackQueue (../../../node_modules/react-dom/cjs/react-dom.development.js:11309:3)
      at scheduleUpdateOnFiber (../../../node_modules/react-dom/cjs/react-dom.development.js:21893:9)
      at dispatchAction (../../../node_modules/react-dom/cjs/react-dom.development.js:16139:5)
      at ../../../node_modules/react-use/lib/useAsyncFn.js:22:61
```

### 👟 Reproduction steps

Start from `npx @backstage/create-app@0.4.37` (latest at the moment) and run tests

### 🖥️ Your Environment

OS:   Linux 5.15.0-60-generic - linux/x64
node: v16.18.1
yarn: 1.22.19
cli:  0.22.2 (installed)
backstage:  1.11.0

Dependencies:
  @backstage/app-defaults                          1.2.0
  @backstage/backend-app-api                       0.4.0
  @backstage/backend-common                        0.18.2
  @backstage/backend-dev-utils                     0.1.0
  @backstage/backend-plugin-api                    0.4.0
  @backstage/backend-tasks                         0.4.3
  @backstage/catalog-client                        1.3.1
  @backstage/catalog-model                         1.2.0
  @backstage/cli-common                            0.1.11
  @backstage/cli                                   0.22.2
  @backstage/config-loader                         1.1.8
  @backstage/config                                1.0.6
  @backstage/core-app-api                          1.5.0
  @backstage/core-components                       0.12.4
  @backstage/core-plugin-api                       1.4.0
  @backstage/errors                                1.1.4
  @backstage/eslint-plugin                         0.1.0
  @backstage/integration-aws-node                  0.1.1
  @backstage/integration-react                     1.1.10
  @backstage/integration                           1.4.2
  @backstage/plugin-api-docs                       0.9.0
  @backstage/plugin-app-backend                    0.3.42
  @backstage/plugin-auth-backend                   0.18.0
  @backstage/plugin-auth-node                      0.2.11
  @backstage/plugin-catalog-backend                1.7.2
  @backstage/plugin-catalog-common                 1.0.11
  @backstage/plugin-catalog-graph                  0.2.27
  @backstage/plugin-catalog-import                 0.9.5
  @backstage/plugin-catalog-node                   1.3.3
  @backstage/plugin-catalog-react                  1.3.0
  @backstage/plugin-catalog                        1.8.0
  @backstage/plugin-github-actions                 0.5.15
  @backstage/plugin-org                            0.6.5
  @backstage/plugin-permission-common              0.7.3
  @backstage/plugin-permission-node                0.7.5
  @backstage/plugin-permission-react               0.4.10
  @backstage/plugin-proxy-backend                  0.2.36
  @backstage/plugin-scaffolder-backend             1.11.0
  @backstage/plugin-scaffolder-common              1.2.5
  @backstage/plugin-scaffolder-node                0.1.0
  @backstage/plugin-scaffolder-react               1.1.0
  @backstage/plugin-scaffolder                     1.11.0
  @backstage/plugin-search-backend-module-pg       0.5.3
  @backstage/plugin-search-backend-node            1.1.3
  @backstage/plugin-search-backend                 1.2.3
  @backstage/plugin-search-common                  1.2.1
  @backstage/plugin-search-react                   1.5.0
  @backstage/plugin-search                         1.1.0
  @backstage/plugin-tech-radar                     0.6.1
  @backstage/plugin-techdocs-backend               1.5.3
  @backstage/plugin-techdocs-module-addons-contrib 1.0.10
  @backstage/plugin-techdocs-node                  1.5.0
  @backstage/plugin-techdocs-react                 1.1.3
  @backstage/plugin-techdocs                       1.5.0
  @backstage/plugin-user-settings                  0.7.0
  @backstage/release-manifests                     0.0.8
  @backstage/test-utils                            1.2.5
  @backstage/theme                                 0.2.17
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by magnusp** [comment]
https://github.com/jsdom/jsdom/issues/3452 for further details.

**Comment by josecosta-shell** [comment]
I just started a vanilla backstage project and this is still an issue.

**Comment by jhaals** [comment]
Hi!

Unfortunately warnings in tests logs is something that do exist today. We want to avoid them if we can and any contributions reduce them in tests logs is highly appreciated.

**Comment by jvilimek** [comment]
upgraded from React 17 to 18 and this bubbled up in our tests... 
package in my app causing this is `jest-environment-jsdom` version `29.0.2` from `@backstage/cli` version `0.25.1`.
Seems to be fixed by [this commit in jest](https://github.com/jestjs/jest/pull/13825/files#diff-66b6c3e7c22db80e40c521842dc614f8a1ee2301de396c3ed6a63d95231ad6c2) and I kind of hoped it would be part of [29.7.0](https://github.com/SimenB/jest/releases/tag/v29.7.0) but its not and the package with (fixed) JSDOM is [v30.0.0-alpha.2](https://github.com/SimenB/jest/releases/tag/v30.0.0-alpha.2) ...
